Design_Doc_Examples/RAG_Q&A_for collaborative_work_platform.md

```diff
@@ -101,20 +101,106 @@ Optional idea to evaluate how good system is at not-clear questions with request
-    3. Metadata plays a vital role in maintaining data consistency and supporting effective data management throughout the lifecycle of an ML system.
-    4. Addressing the cold start problem requires creative approaches to data acquisition and utilization, emphasizing the need for flexibility and innovation in early system development stages.
-    5. Establishing a healthy data pipeline is foundational to the long-term success and scalability of ML systems, underscoring the importance of reproducibility, consistency, and availability in data management practices.
+We don't control the data for the main LLM training (meaning that we're coping with the LLM limitations and don't influence this until the moment we realize that we could really benefit from fine-tuning, which will become completely different project).
+
+#### i. Data to perform RAG on - description
+
+We don't distinguish between client roles for data access. Every client would have access to every document. So we basically have a shared dataset for the whole system.
+
+Includes a set of Documents available on the Platform. Documents can be in text format (Markdown) or scanned/image formats.
+
+1. For Markdown documents
+    - Expected Document Size: Up to 500 pages.
+    - Structure: Documents larger than 10 pages typically include a table of contents and dedicated sections, such as introduction or glossary.
+    - Content: Documents may include text with all Markdown features (e.g., quotes, headings, formulas, tables).
+2. For documents in Image form (scans/images): no additional description available. They are just files that contain image/scan inside.
+3. Each document have origination metadata
+4. Documents may have v1-v2-v3-... versions. For some Documents we know the diff between vX and vY. For some Documents we do not know the diff, and only know how Document looked like at each version.
+
+Clients can edit documents online via the platform or upload documents from their local machines. Each document receives a version number upon:
+
+- Saving
+- Uploading a new document
+- Uploading a version of the existing document
+
+Clients can access all versions of each document.
+
+#### ii. Data cleaning
+
+Data cleaning process should be automatized with reproducible scripts. They need to run with some schedule, regularly + on huge new document upload.
+Question: is version control needed? We already have one in the system. How to deal with that?
+
+**1. Markdown Documents:**
+- Links: do we need to save links? If yes, check that links within the document are not broken. Update or remove broken links.
+- Duplicate removal. Need to ensure that different versions of one document are not treated as duplicates despite being very similar.
+- Table of Contents (ToC): Generate or validate the presence of a ToC for documents longer than 10 pages. Tools like Pandoc or custom scripts can automate this.
```

Comment:
We may enrich our text by recognised ToC.
Or somehow use explicit ToC during constructing Prompt

Reply:
Yes, this is now covered in metadata section